custom_components/dahua/client.py

```diff
@@ -87,22 +101,36 @@ async def get_device_type(self) -> dict:
         Some cams might return...
         type=IP Camera
         """
-        return await self.get("/cgi-bin/magicBox.cgi?action=getDeviceType")
+        try:
+            return await self.get("/cgi-bin/magicBox.cgi?action=getDeviceType")
+        except aiohttp.ClientResponseError as e:
+            return {"type":"Generic RTSP"}
 
     async def get_software_version(self) -> dict:
         """
         get_software_version returns the device software version (also known as the firmware version). Example response:
         version=2.800.0000016.0.R,build:2020-06-05
         """
-        return await self.get("/cgi-bin/magicBox.cgi?action=getSoftwareVersion")
+        try:
+            return await self.get("/cgi-bin/magicBox.cgi?action=getSoftwareVersion")
+        except aiohttp.ClientResponseError as e:
+            return {"version":"1.0"}
 
     async def get_machine_name(self) -> dict:
         """ get_machine_name returns the device name. Example response: name=FrontDoorCam """
-        return await self.get("/cgi-bin/magicBox.cgi?action=getMachineName")
+        not_hashed_id = "{0}_{1}_{2}_{3}".format(self._address, self._rtsp_port, self._username, self._password)
+        unique_cam_id = md5((not_hashed_id).encode('UTF-8')).hexdigest()
+        try:
+            return await self.get("/cgi-bin/magicBox.cgi?action=getMachineName")
+        except aiohttp.ClientResponseError as e:
+            return {"name":unique_cam_id}
```

Comment:
Can you leave a comment here what this is. Any context would help

Reply:
Dahua camera return unique identifier by "/cgi-bin/magicBox.cgi?action=getMachineName". For RTSP camera need to generate unique id ad use this id everywhere for identify objects. File core.device_registry provide json information like this

 {
                "config_entries": [
                    "374e8200431dc735791ae0f63556d781",
                    "100389553d3d6524d8c73dde2e474d00"
                ],
                "connections": [
                    [
                        "mac",
                        "00:30:6f:83:d1:21"
                    ]
                ],
                "identifiers": [
                    [
                        "keenetic_ndms2",
                        "00:30:6f:83:d1:21"
                    ],
                    [
                        "dahua",
                        "8026a73660b0371a68cfe9169ca6830d"
                    ]
                ],
                "manufacturer": "Dahua",
                "model": "Generic RTSP",
                "name": "RTSP_Camera_1",
                "sw_version": "1.0",
                "hw_version": null,
                "entry_type": null,
                "id": "32777735d851d72aa73a8c6e65b33597",
                "via_device_id": null,
                "area_id": "ulitsa",
                "name_by_user": null,
                "disabled_by": null,
                "configuration_url": "http://192.168.17.11:554/cam0_0"
            },